trl/trainer/moreh_utils.py

```diff
@@ -37,14 +33,17 @@ def on_log(self, args: TrainingArguments, state: TrainerState, control: TrainerC
                 mlflow.log_metric("throughput", throughput , step=state.global_step)
                 mlflow.log_metric("train_loss", state.log_history[-1]["loss"] , step=state.global_step)
                 print(f'loss: {state.log_history[-1]["loss"]}, lr: {state.log_history[-1]["learning_rate"]}, throughput: {throughput}, step: {state.global_step}')
-
-            # when an epoch ends
             
     def on_epoch_end(self, args: TrainingArguments, state: TrainerState, control: TrainerControl, **kwargs):
-        # calculate average loss of epoch
-        epoch_loss = sum(self.epoch_loss_list) / len(self.epoch_loss_list)
-        mlflow.log_metric("epoch_loss", epoch_loss, step=state.epoch)
-        self.epoch_loss_list = []
+        all_loss_of_this_epoch = []
+        for each in state.log_history[self.epoch_start:]:
+            if 'loss' in each:
+                all_loss_of_this_epoch.append(each['loss'])
+
+        epoch_loss = sum(all_loss_of_this_epoch) / len(all_loss_of_this_epoch)
+        mlflow.log_metric("epoch_loss", epoch_loss, step=int(state.epoch))
```

Comment:
How can it be treated differently? Its step number is epoch-based while `on_log` takes global_step for step.

Reply:
@yeomni the "step" arg in mlflow isnt the same as step in training, it's the number on the x-axis signifying the space between values. Since here I'm logging to mlflow after each epoch, the step is the same as epoch number